lib/ably/modules/conversions.rb

```diff
@@ -115,5 +115,38 @@ def ensure_supported_payload(payload)
 
       raise Ably::Exceptions::UnsupportedDataType.new('Invalid data payload', 400, Ably::Exceptions::Codes::INVALID_MESSAGE_DATA_OR_ENCODING)
     end
+
+    # It converts the name, data, attributes into the array of Message objects
+    #
+    # @return [Array<Ably::Models::Message>]
+    #
+    def build_messages(name, data = nil, attributes = {})
+      if name.kind_of?(Enumerable)
```

Comment:
Since the only problem here is to support both array and single objects, I think it may be worthwhile to consider the following construction 

 ```ruby
def build_messages(name, data = nil, attributes = {})
  Array(name)e.map do |item|
    Ably::Models::Message(ensure_supported_name_and_payload(item, data, attributes).dup)
  end
end
```

It achieves the same thing and reads better. wdyt?

Reply:
Looks better but the above construction does not work.

For

```ruby
build_messages(nil, 'example data')
```

it should return

```ruby
[{"data": "example data"}]

```

but it returns empty array `[]`


`8` tests fail (most of them related to the `nil` name):

```spec

3101 examples, 8 failures, 10 pendings

Took 2052 seconds (34:12)
Tests Failed
\e[31m⚠ Note: Unit tests have also failed, but are not listed in the test failures above. Scroll up to the unit tests ⚠\e[0m
Unit tests exit code: 1
Acceptance tests exit code: 1

```

Reply:
Fixed b4acff6992ccc69f1ee74ca5b336f7986d7e95da